server/base/src/main/java/org/apache/accumulo/server/conf/util/ZooInfoViewer.java

```diff
@@ -0,0 +1,540 @@
+      }
+
+      writer.println("-----------------------------------------------");
+    }
+  }
+
+  /**
+   * Get the instanceID from the command line options, or from value stored in HDFS. The search
+   * order is:
+   * <ol>
+   * <li>command line: --instanceId option</li>
+   * <li>command line: --instanceName option</li>
+   * <li>HDFS</li>
+   * </ol>
+   *
+   * @param zooReader
+   *          a ZooReader
+   * @param opts
+   *          the parsed command line options.
+   * @return an instance id
+   */
+  InstanceId getInstanceId(final ZooReader zooReader, final ZooInfoViewer.Opts opts) {
+
+    if (!opts.instanceId.isEmpty()) {
+      return InstanceId.of(opts.instanceId);
+    }
+    if (!opts.instanceName.isEmpty()) {
+      Map<String,InstanceId> instanceNameToIdMap = readInstancesFromZk(zooReader);
+      String instanceName = opts.instanceName;
+      for (Map.Entry<String,InstanceId> e : instanceNameToIdMap.entrySet()) {
+        if (e.getKey().equals(instanceName)) {
+          return e.getValue();
+        }
+      }
+      throw new IllegalArgumentException(
+          "Specified instance name '" + instanceName + "' not found in ZooKeeper");
+    }
+
+    try (ServerContext context = new ServerContext(SiteConfiguration.auto())) {
+      return context.getInstanceID();
+    } catch (Exception ex) {
+      throw new IllegalArgumentException(
+          "Failed to read instance id from HDFS. Instances can be specified on the command line",
+          ex);
+    }
+  }
+
+  Map<NamespaceId,String> getNamespaceIdToNameMap(InstanceId iid, final ZooReader zooReader) {
+    SortedMap<NamespaceId,String> namespaceToName = new TreeMap<>();
+    String zooNsRoot = ZooUtil.getRoot(iid) + ZNAMESPACES;
+    try {
+      List<String> nsids = zooReader.getChildren(zooNsRoot);
+      for (String id : nsids) {
+        String path = zooNsRoot + "/" + id + ZNAMESPACE_NAME;
+        String name = new String(zooReader.getData(path), UTF_8);
+        namespaceToName.put(NamespaceId.of(id), name);
+      }
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException("Interrupted reading namespace ids from ZooKeeper", ex);
+    } catch (KeeperException ex) {
+      throw new IllegalStateException("Failed to read namespace ids from ZooKeeper", ex);
+    }
+    return namespaceToName;
+  }
+
+  private void printProps(final InstanceId iid, final ZooReader zooReader, final Opts opts,
+      final PrintWriter writer) {
+
+    if (opts.printAllProps()) {
+      log.info("all: {}", opts.printAllProps());
+    } else {
+      log.info("Filters:");
+      log.info("system: {}", opts.printSysProps());
+      log.info("namespaces: {} {}", opts.printNamespaceProps(),
+          opts.getNamespaces().size() > 0 ? opts.getNamespaces() : "");
+      log.info("tables: {} {}", opts.printTableProps(),
+          opts.getTables().size() > 0 ? opts.getTables() : "");
```

Comment:
The default log configuration will print these to the console. These log messages will be interspersed with the non-log STDOUT. By default, though, they will go to STDERR. That may be what you want (so you can pipe them to another command without the extra info)... but if you intend them to always be included in the output, they should use the same console writer as the printing of the values.

The other thing here is that you have multiple log statements in a row. This seems a bit unnecessary... like you're using logging for a non-logging purpose.

Reply:
I think it might be better to try to make the log messages more succinct, rather than try to format their output inline with the other output. I can try to think on it and come up with a proposed change.

Also, I was thinking about suggesting a more computer-friendly output format for follow-on processing. Similar to GPG's `--with-colons`, or like the `--short` option for `git status`.

Having more options, like only printing the specific configuration being asked, could also help with the output... less information is needed to organize the output if you are only getting one table or namespace or system scope at a time.

Reply:
One thing to consider is that this is geared towards emergency use and while it is really helpful to have the information, its usage is going to be situational and highly dependent on what needs to be done.  For a while I had a json output option (for properties) but it seemed to add little utility.  With the current output it is structured enough that grep / awk can generate shell command(s) without much issue.  The most common cases that I thought of:

 - Dump a table's property to setup a mirror system
 - Dump the id mapping to recover from issues where a bulk re-import is the only option.

The sticking point for me was that sometimes knowing the id is really important - but ids are specific to an instance and have no utility across instances.  

I'm not opposed to other options - just relaying things that I had considered in case that shapes any suggestions.

Reply:
```
Having more options, like only printing the specific configuration being asked, could also help with the output... less information is needed to organize the output if you are only getting one table or namespace or system scope at a time.
```

If you specify a filter - you only receive output for those items.  For example
```
accumulo zoo-info-viewer --instanceName uno --print-props -t tbl1
```
# Sample output (single table)
<details>
  <summary>Click to expand!</summary>
  
  ```
-----------------------------------------------
Report Time: 2022-06-03T10:42:05.718655Z
-----------------------------------------------
ZooKeeper properties for instance ID: 33ac4ded-7ead-436b-983e-401e50464190

Tables: 
Name: tbl1, Data Version:1, Data Timestamp: 2022-06-03T10:37:29.810437Z:
  table.constraint.1=org.apache.accumulo.core.data.constraints.DefaultKeySizeConstraint
  table.iterator.majc.vers=20,org.apache.accumulo.core.iterators.user.VersioningIterator
  table.iterator.majc.vers.opt.maxVersions=1
  table.iterator.minc.vers=20,org.apache.accumulo.core.iterators.user.VersioningIterator
  table.iterator.minc.vers.opt.maxVersions=1
  table.iterator.scan.vers=20,org.apache.accumulo.core.iterators.user.VersioningIterator
  table.iterator.scan.vers.opt.maxVersions=1


-----------------------------------------------
  ```
</details>

The choice of providing name=value form was deliberate.  1) It allows an easy filter and also matches the syntax for the shell `config` command.  So, piping the command output to something like ` | grep '=' | tr '\n''  ` is just about the format needed to make a shell config command.